### ISSUE TITLE
DeviceClient 1.39.0 hangs on close using WinUi3 desktop application

### ISSUE THREAD

**Issue by mavado83** [Azure/azure-iot-sdk-csharp#2269]
Using Microsoft.Azure.Devices.Client version 1.39.0 in a WinUi 3 desktop app targeting framework => net6.0-windows10.0.19041.0. 
When closing the application, the MqTT connection stays open.

```
 _deviceClient =  DeviceClient.CreateFromConnectionString(_options.DeviceConnectionString, TransportType.Mqtt,  options);
```
Switched back to version 1.35.0 and the problem does not occur.

**Comment by abhipsaMisra** [comment]
Linked #2163 , #2194

**Comment by jamdavi** [comment]
We had been waiting on a release from the Netty team and discussed internally the best way to handle the update. The Netty team created a new version of the Common library and we are now taking a dependency on this new version of the library.


**This is not required once we release the new SDK** but  if you want to test this out you can also take a dependency on the latest version of DotNetty.Common and it should replace the assembly at build time.

Note, if you're targeting .NET Full framework such as net472 you'll have to use a binding redirect in your app.config. Visual Studio has an option to generate one automatically.

**Comment by mavado83** [comment]
I updated to Azure Client 1.39.0 and referenced DotNetty.Common 0.7.1. I can confirm that the connection is closed as it supposed to be.

**Comment by jamdavi** [comment]
This is now fixed with version 1.40.0 [https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2022_01_26].
